src/Algolia.Search.Test/Serializer/SerializerTest.cs

```diff
@@ -198,23 +198,23 @@ public void TestAutomaticFacetFilters()
         [Parallelizable]
         public void TestLegacyFilterFormats()
         {
-            // Testing "one string" legacy filters => should be converted to "ORED" nested filters
-            // [["color:green","color:yellow"]]
+            // Testing "one string" legacy filters => should be converted to "ANDED" nested filters
+            // [["color:green"],["color:yellow"]]
             string stringFilters = "\"color:green,color:yellow\"";
 
             var serializedStringFilters =
                 JsonConvert.DeserializeObject<List<List<string>>>(stringFilters, new FiltersConverter());
 
-            AssertOredResult(serializedStringFilters);
+            AssertAndedResult(serializedStringFilters);
 
-            // Testing "one array" legacy filters => should be converted to "ORED" nested filters
-            // [["color:green","color:yellow"]]
+            // Testing "one array" legacy filters => should be converted to "ANDED" nested filters
+            // [["color:green"],["color:yellow"]]
             string arrayFilters = "[\"color:green\",\"color:yellow\"]";
```

Comment:
I'd also add a test with `"["color:green,color:blue","color:yellow"]";` IIUC it's the only case we are not testing.

Reply:
👋🏻 @godelized - just to confirm, Is what I proposed a valid filter format?

Reply:
mmh actually @sbellone I think it is not supported:

```
filters: Unexpected token string(color:green,color:blue) expected end of filter at col 3.
```

Reply:
Ok so I doubled checked with @godelized and the expected output is:

```
["color:green,color:blue","color:yellow"] <=> [["color:green,color:blue"],["color:yellow"]]
```

Reply:
In other words when the input is the JSON array, there's no split of the strings.

Reply:
Perfect, that's the current behaviour, just adding a test then :+1:

Reply:
Ok actually my change had broken this behaviour. All cases should be good now